### PR TITLE
fix: resolve image caching issue on flame_game template

### PR DESCRIPTION
**Pull request by BBarisKilic**
## Status

**READY**

## Description

Fixes: https://github.com/VeryGoodOpenSource/very_good_cli/issues/918

There is a bug on the generated flame_game template. The cached images are not accessible from the FlameGame, therefore they are not "actually" cached and causing a runtime error. This pull request fixes the issue.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: src/very_good_flame_game/test/game/components/counter_component_test.dart [hunk 1]
```diff
@@ -18,29 +19,34 @@ class _VeryGoodFlameGame extends VeryGoodFlameGame {
...
-  _VeryGoodFlameGame createFlameGame() {
+  late final AppLocalizations l10n;
+
+  setUpAll(() {
+    l10n = _MockAppLocalizations();
+
+    when(() => l10n.counterText(any())).thenAnswer(
+      (invocation) => 'counterText: ${invocation.positionalArguments[0]}',
+    );
+  });
```
Comment (wolfenrain) [inline src/very_good_flame_game/test/game/components/counter_component_test.dart:38]: We should move this back into the group, and use `setUp` so each test gets a fresh mock.

File: src/very_good_flame_game/test/game/entities/unicorn/unicorn_test.dart [hunk 2]
```diff
@@ -28,20 +34,27 @@ class _VeryGoodFlameGame extends VeryGoodFlameGame {
...
+  late final AppLocalizations l10n;
+  late final Images images;
+
+  setUpAll(() {
+    l10n = _MockAppLocalizations();
+    images = _MockImages();
+
+    when(() => l10n.counterText(any())).thenReturn('counterText');
+    when(() => images.fromCache(any())).thenReturn(_FakeImage());
+  });
```
Comment (wolfenrain) [inline src/very_good_flame_game/test/game/entities/unicorn/unicorn_test.dart:46]: This should go into the group and be `setUp` so each test gets it's own mocked info

**Comment by alestiago** [comment]
Resolves [#918](https://github.com/VeryGoodOpenSource/very_good_cli/issues/918)

File: src/very_good_flame_game/test/game/components/counter_component_test.dart (hunk 1 above)
Comment (BBarisKilic) [reply src/very_good_flame_game/test/game/components/counter_component_test.dart]: Done.

File: src/very_good_flame_game/test/game/entities/unicorn/unicorn_test.dart (hunk 2 above)
Comment (BBarisKilic) [reply src/very_good_flame_game/test/game/entities/unicorn/unicorn_test.dart]: Done.

**Comment by BBarisKilic** [comment]
Tests have been updated as @wolfenrain suggested.

**Review by wolfenrain — APPROVED** [review]

**Pull request merged**